samtranslator/model/apigateway.py

```diff
@@ -270,8 +270,22 @@ def _is_missing_identity_source(self, identity):
         query_strings = identity.get("QueryStrings")
         stage_variables = identity.get("StageVariables")
         context = identity.get("Context")
+        ttl = identity.get("ReauthorizeEvery")
 
-        if not headers and not query_strings and not stage_variables and not context:
+        required_properties_missing = not headers and not query_strings and not stage_variables and not context
+
+        try:
```

Comment:
I think that the reason we reverted was because of an intrinsic function in `ttl`.
If `ttl` is an intrinsic, I think we should return `False` (no access to the value, assume it's valid, pass it to CFN).
We can use the `is_intrinsic` function to test this.

Then we can use `six`, `integer_types`:
`from six import integer_types`
`return (!isinstance(ttl, integer_types) or ttl == 0) and required_properties_missing`

Reply:
Thanks for the feedback. `try: except:` is pretty normal in python and I prefer this over checking instances types. "Pythonic" style tends to be do the thing and then handle errors. `try: except:` is not slower, which tends to be a reason to avoid it in other languages, https://stackoverflow.com/questions/7604636/better-to-try-something-and-catch-the-exception-or-test-if-its-possible-first has some good explanations. 

With that I prefer the `try: except:` implementation over the checking instance.

I am generally cautious around doing multiple things within a `try` block. If something is changed in the future, you run the risk of the `except` catching on unknown things. Or even if something else unexpected happens, the `except` can mask things. So well scoped `try` tends to be a better practice. I will see how I can simplify this further, but pretty strongly against over-scoping `try` blocks.